Automattic/wp-calypso
# disable upgrade to p2+, only show plan if it is the current plan

Related to #

## Proposed Changes

* Hide the P2+ plan from the plans view, unless it is the active plan
* additionally set `availableForPurchase` flag to `false` incase it is used anywhere else

| before | after | after if plan purchased |
| -- | -- | -- |
| <img width="788" alt="SCR-20240402-pjmi" src="https://github.com/Automattic/wp-calypso/assets/4081020/f0a991e2-2923-4b4b-a946-2ba496dea007"> | <img width="595" alt="SCR-20240402-pjoe" src="https://github.com/Automattic/wp-calypso/assets/4081020/fb9b8333-3630-41be-b219-b02f8e59afe2"> | <img width="835" alt="SCR-20240402-pkgf" src="https://github.com/Automattic/wp-calypso/assets/4081020/98b28334-cb30-410f-9397-df9ccf465eb7"> |

## Testing Instructions



* pull pr
* `yarn && yarn start`
* You MIGHT need to also build the `plans-grid-next` package: `yarn workspace @automattic/plans-grid-next run prepare`
* create a new personal P2 site from https://p2.blog not an automattic p2 site (or use an existing one)
* try to view the Plans page after creation, the P2+ plan should not appear as an option
* add P2+ via store-admin to your P2 hub site
* view the Plans page again, P2+ should appear because it is the current plan

## Pre-merge Checklist
